core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,43 +330,86 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
+    // doesn't have a review message. When a contribution has
+    // a review message the second element of the threadMessages
+    // array contains the actual review message.
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
     }
+  }
 
-    let lastContribution = (
-      Object.keys(this.remainingContributions)[
-        Object.keys(this.remainingContributions).length - 1]);
-    this.activeSuggestionId = lastContribution;
-    this.activeContribution = this.remainingContributions[
-      lastContribution];
-    delete this.remainingContributions[this.activeSuggestionId];
+  refreshModalData(): void {
     // Close modal instance if the suggestion's corresponding opportunity
     // is deleted. See issue #14234.
     if (!this.activeContribution.details) {
       this.activeModal.close(this.resolvedSuggestionIds);
       return;
     }
-
-    this.activeSuggestion = this.activeContribution.suggestion;
     this.activeContributionDetails = this.activeContribution.details;
+    this.activeSuggestion = this.activeContribution.suggestion;
+    if (this.activeContribution.details === null) {
+      return;
+    }
     this.contextService.setCustomEntityContext(
       AppConstants.IMAGE_CONTEXT.EXPLORATION_SUGGESTIONS,
       this.activeSuggestion.target_id);
     this.subheading = (
-      this.activeContributionDetails.topic_name + ' / ' +
-      this.activeContributionDetails.story_title +
-      ' / ' + this.activeContributionDetails.chapter_title);
+      `${this.activeContribution.details.topic_name} / ` +
+      `${this.activeContribution.details.story_title} / ` +
+      `${this.activeContribution.details.chapter_title}`
+    );
+  }
+
+  goToNextItem(): void {
+    const lastContributionId = this.remainingContributionIds.pop();
+    // If the current item is the last item, do not navigate.
+    if (lastContributionId === undefined) {
+      return;
+    }
+    // This prevents resolved contributions from getting added to the list.
+    if (!this.resolvedSuggestionIds.includes(this.activeSuggestionId)) {
+      this.skippedContributionIds.push(this.activeSuggestionId);
+    }
+
+    this.activeSuggestionId = lastContributionId;
+
     this.init();
   }
 
+  goToPreviousItem(): void {
+    const lastContributionId = this.skippedContributionIds.pop();
+    // If the current item is the first item, do not navigate.
+    if (lastContributionId === undefined) {
+      return;
+    }
+    // This prevents resolved contributions from getting added to the list.
+    if (!this.resolvedSuggestionIds.includes(this.activeSuggestionId)) {
+      this.remainingContributionIds.push(this.activeSuggestionId);
+    }
+
+    this.activeSuggestionId = lastContributionId;
+
+    this.init();
+  }
+
+  resolveSuggestionAndUpdateModal(): void {
+    this.resolvedSuggestionIds.push(this.activeSuggestionId);
+
+    // Remove resolved contributions from the record.
```

Comment:
```suggestion
    // Resolved contributions don't need to be displayed in the modal.
```

More to explain the "why".